### PR TITLE
Make `faq.md` more relevant to the current state of the project

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -38,29 +36,73 @@ hash after their name, then this unexpected. Check to see if your build is
...
+library dependency is also directly depended on by another top level target,
+such as `ios_unit_test`, then we can't merge that library into the first top
+level target. When that happens, the first top level target doesn't have any
+source files, so we need to add a stub one to allow Xcode to link to the proper
+library target.
+
+If this setup isn't desired (e.g. wanting to have the target merged to enable
+SwiftUI Previews), there are a couple ways to fix it. For tests, setting the
+first top level target as the `test_host` will allow for the library to merge.
+In other cases, refactor the build graph to have the shared code in it's own
```
Comment (jpsim) [inline docs/faq.md:51]: Nit

```suggestion
In other cases, refactor the build graph to have the shared code in its own
```

**Review by jpsim — APPROVED** [review]

File: docs/faq.md [hunk 2]
```diff
@@ -38,29 +36,73 @@ hash after their name, then this unexpected. Check to see if your build is
...
+- Remove the use of header maps (hmaps), or don't include Swift generated
+  headers in them
+- Use BwB mode instead
 
-All targets that depend on generates files depend on the `Bazel Generated Files`
-target, so building any of those targets will also synchronize Xcode.
+## Why does Xcode complain about stale files?
 
-## When I build I get warnings like "Stale file 'PROJECT.xcodeproj/rules_xcodeproj/gen_dir/...' is located outside of the allowed root paths". How do I fix this?
+The generated project tracks Bazel managed external and generated files. If you
```
Comment (brentleyjones) [inline docs/faq.md:71]: > The generated project tracks Bazel managed external and generated files.

I just realize that BwB mode probably doesn't need to do this anymore. I'll add a TODO to explore this.

File: docs/faq.md [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
-* [Does the Archive action work?](#does-the-archive-action-work)
+## What do BwB and BwX mean?
+
+The `xcodeproj` rule supports two different
+[build modes](bazel.md#xcodeproj-build_mode), `"bazel"` and `"xcode"`, commonly
+referred to as Build with Bazel (BwB) and Build with Xcode (BwX).
+
+## Does the Archive action work?
+
+Currently no effort is put into making the Archive action work correctly.
```
Comment (brentleyjones) [inline docs/faq.md:11]: I wonder if we should prevent the archive action from being enabled in all schemes to make this more clear?

File: docs/faq.md [hunk 4]
```diff
@@ -38,29 +36,73 @@ hash after their name, then this unexpected. Check to see if your build is
...
+The generated project tracks Bazel managed external and generated files. If you
+remove a dependency on one of those files, or the path of one of those files
+changes (e.g. because of a Bazel configuration change), and generate the project
+without closing and reopening it, or performing a clean build, Xcode will warn
+about the old paths. This seems to be a caching bug in Xcode.
 
-This warning happens when the paths to generated files change after a project
-generation. Closing and re-opening your project, or performing a clean build,
-will remove the warnings.
+## Do I need to place my custom Xcode scheme declarations in a function like `tools/generator`?
```
Comment (brentleyjones) [inline docs/faq.md:77]: I believe with Bazel 6.1 this wrinkle should go away, as https://github.com/bazelbuild/bazel/issues/17260 will hopefully be able to be called off of a `BUILD` thread.

**Pull request merged**